ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/CommandClient.scala

```diff
@@ -61,9 +56,14 @@ final class CommandClient(
   def submitSingleCommand(
       submitRequest: SubmitRequest,
       token: Option[String] = None): Future[Empty] =
+      submit(token)(submitRequest)
+
+  private def submit(token: Option[String])(submitRequest: SubmitRequest): Future[Empty] = {
+    logger.debug(s"Invoking grpc-submission on commandId=${submitRequest.commands.map(_.commandId).getOrElse("no-command-id")}")
```

Comment:
```suggestion
    logger.debug("Invoking grpc-submission on commandId={}", submitRequest.commands.map(_.commandId).get)
```
1. The commands will be there, no need to have a fallback value, if it passed validation there's a bug
2. When using `logback` directly, favor `logback`'s templates over Scala string interpolation (the former acts lazily)

Reply:
1. But this is a client library. I'm sure that clients will mess this up and failing with a random scala exception on a log message would be shitty. So I'd prefer to keep it.
2. You are right with the `logback`. Fixed.